src/main/java/ryze/Ryze.java

```diff
@@ -0,0 +1,380 @@
+    private static final String EXIT_COMMAND = "bye";
+    private static final String LIST_COMMAND = "list";
+    private static final String TODO_COMMAND = "todo";
+    private static final String DEADLINE_COMMAND = "deadline";
+    private static final String EVENT_COMMAND = "event";
+    private static final String MARK_COMMAND = "mark";
+    private static final String UNMARK_COMMAND = "unmark";
+    private static final String DELETE_COMMAND = "delete";
+    private static final String TODO = "T";
+    private static final String DEADLINE = "D";
+    private static final String EVENT = "E";
+
+
+
+    private static ArrayList<Task> listOfChatHistory = new ArrayList<>();  // Static task list
+
+    public static void main(String[] args) {
+        initialiseData();
+        initialiseMessage();
+
+        Scanner scanner = new Scanner(System.in);
+        String line;
+        int lastSize = listOfChatHistory.size();
+        while (!(line = scanner.nextLine()).equals(EXIT_COMMAND)) {
+            processCommand(line);
+            lastSize = handleSizeChange(lastSize);
+        }
+
+        exitMessage();
+
+
+    }
+
+    private static int handleSizeChange(int lastSize) {
+        int currentSize = listOfChatHistory.size();
+        if (lastSize > currentSize){
+            try{
+                overwriteSave();
+            } catch (IOException e) {
+                System.out.println("Error overwriting save file" + e.getMessage());
+            }
+        }
+        lastSize = currentSize;
+        return lastSize;
+    }
+
+    private static void initialiseData() {
+        Path dirPath = Paths.get("data");
+        Path filePath = dirPath.resolve("ryze.txt");
+        String filePathString = filePath.toString();
+        File file = new File(filePathString);
+        System.out.println(filePathString);
+
+        // Check if directory exists, if not, create it
+        if (!Files.exists(dirPath)) {
+            try {
+                Files.createDirectories(dirPath);
+            } catch (IOException e) {
+                System.out.println("Error creating directory: " + e.getMessage());
+                return;
+            }
+        }
+
+        // Check if file exists, if not, create it
+        if (!file.exists()) {
+            System.out.println("Data for Ryze not found! Welcome to Ryze new User!");
+            try {
+                Files.createFile(filePath); // Creates the file if it doesn't exist
+            } catch (IOException e) {
+                System.out.println("Something went wrong: " + e.getMessage());
+            }
+        } else {
+            try {
+                readFileContents(filePathString);
+            } catch (IOException e) {
+                System.out.println("Error reading file: " + e.getMessage());
+            }
+        }
+    }
+
+    private static void appendToFile(String textToAppend) throws IOException {
+        Path dirPath = Paths.get("data");
```

Comment:
You may consider extracting "data" into a constant as well since it's used multiple times in the code.